### ISSUE TITLE
Inconsistent behaviour when uploading to external storage root vs sub directory

### ISSUE THREAD

**Issue by st3iny** [nextcloud/files_accesscontrol#605]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

#### Preparation (one time)
1. Configure a file access rule to block uploads above 5 Mb
1. Ensure that chunked upload is enabled: `occ config:app:delete files max_chunk_size`
1. Add an external S3 storage.
1. Create a sub directory inside the external storage.
#### Case 1 (correct)
1. Upload a big file directly to the external storage's root folder.
2. Observe that the upload is blocked **after** all chunks were uploaded (error on assembly).
#### Case 2 (broken)
1. Upload a big file to the sub directory inside the external store.
2. Observe that the upload is blocked **immediately** when starting the first chunk.

### Expected behavior
Case 1 is behaving as expected. Case 2 is **not** behaving as expected. The upload should be blocked after assembly in both cases.

### Actual behavior
The upload is blocked **immediately** in case 2 instead of postponing the check until the whole file is available like in case 1.

### Server configuration

**Web server:** Apache (`content_length` fix is applied)

**PHP version:** 8.1

**Nextcloud version:** master

**Comment by nickvergessen** [comment]
> Case 1 (correct)
> Case 2 (broken)

Not sure about that. As soon as we know that the upload will be blocked, we should block it.

**Comment by st3iny** [comment]
True, but now I'm really confused. Weren't the checks delayed on purpose to prevent early scans on incomplete files? Because some checks might yield false positives when run on chunks instead of the complete file.

So I'd expect consistent behavior of waiting until all chunks are uploaded for all cases. Regardless of the file is being uploaded to `/my-ext-storage` or `/my-ext-storage/foobar/`.

See https://github.com/nextcloud/files_accesscontrol/pull/330
